src/unix/hermit/mod.rs

```diff
@@ -19,10 +19,8 @@
 pub type c_long = i64;
 pub type c_ulong = u64;
 
-#[allow(unused)]
-pub type uid_t = u16;
-#[allow(unused)]
-pub type gid_t = u16;
+//pub type uid_t = u16;
```

Comment:
My point here was that if rustc says that these are unused, is because you can't access them via `libc::uid_t`. This might hint that the contents of the `hermit` module are not properly exported by the root crate.

Reply:
You are right. At export the module in [src/unix/mod.rs](https://github.com/rust-lang/libc/blob/ff97bdb074782fc5bbcae88c405d408672726f3d/src/unix/mod.rs#L1146-L1148). For me isn't clear, why I got these warnings.

Reply:
These are also defined in the general `unix/mod.rs`: https://github.com/rust-lang/libc/blob/ff97bdb074782fc5bbcae88c405d408672726f3d/src/unix/mod.rs#L38

So they are probably being shadowed, and therefore not re-exported (hence the `unused` warning - these are dead code).

Reply:
Basically you'd need to move `gid_t` and `uid_t` from `unix/mod.rs` to all other sub-modules of the `unix/` directory.

Reply:
Ok, I oversaw this. `u32` is also fine for me. I will delete these types and use `u32` in our lib.